src/diffpy/utils/diffraction_objects.py

```diff
@@ -390,14 +390,15 @@ def scale_to(self, target_diff_object, xtype=None, xvalue=None):
 
         data = self.on_xtype(xtype)
         target = target_diff_object.on_xtype(xtype)
+        if len(data[0]) == 0 or len(target[0]) == 0 or len(data[0]) != len(target[0]):
+            raise ValueError("I cannot scale two diffraction objects with empty or different lengths.")
         if xvalue is None:
             xvalue = data[0][0] + (data[0][-1] - data[0][0]) / 2.0
 
         xindex = (np.abs(data[0] - xvalue)).argmin()
         ytarget = target[1][xindex]
         yself = data[1][xindex]
```

Comment:
Does it make more sense that we find an xindex for the original DO that we want to scale, and then another xindex for the target DO, and scale based on that?